test/lib/tools/AndByDiscriminatingKey.type.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { AndByDiscriminatingKey } from "keycloakify/tools/AndByDiscriminatingKey";
+import { assert } from "tsafe/assert";
+import type { Equals } from "tsafe";
+import { it, describe } from "vitest";
+
+// These test case names are intentionally vague, because I'm not sure what each are testing individually
+describe("AndByDiscriminatingKey", () => {
+    it("Test Case 1", () => {
```

Comment:
Should update the test cases in this file

Reply:
I don't mind the file being renames but I would avoid having type level test files registered to be ran by the test runner.  
This test is not mean to be ran. If it compiles it passes. We just need to make sure this test is tracked by tsconfig.  
I'll handle it!